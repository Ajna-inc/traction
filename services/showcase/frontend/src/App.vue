<template>
  <v-app>
    <BaseNotificationContainer />
    <AppHeader />
    <AppNavBar />

    <v-main>
      <transition name="component-fade" mode="out-in">
        <router-view />
      </transition>
    </v-main>

    <AppFooter />
  </v-app>
</template>

<script>
import AppHeader from '@/components/app/AppHeader.vue';
import AppFooter from '@/components/app/AppFooter.vue';
import AppNavBar from '@/components/app/AppNavBar.vue';

export default {
  name: 'App',
  components: {
    AppHeader,
    AppFooter,
    AppNavBar
  }
};
</script>

<style scoped>
.component-fade-enter-active,
.component-fade-leave-active {
  transition: opacity 0.3s ease;
}
.component-fade-enter,
.component-fade-leave-to {
  opacity: 0;
}
</style>
