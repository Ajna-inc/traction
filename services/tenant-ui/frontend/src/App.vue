<script setup lang="ts">
import ConfirmPopup from 'primevue/confirmpopup';
import AppLayout from './components/layout/AppLayout.vue';
import Login from './components/Login.vue';
import { onMounted } from 'vue';
import { storeToRefs } from 'pinia';
import { useConfigStore, useTenantStore } from './store';

const { config } = storeToRefs(useConfigStore());
const { tenantReady } = storeToRefs(useTenantStore());

onMounted(() => {
  document.title = config.value.ux.appTitle;
});
</script>

<template>
  <AppLayout v-if="tenantReady" />
  <Login v-else />

  <!-- Shared confirm popup  -->
  <ConfirmPopup />
</template>

<style scoped>
#app {
  height: 100%;
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
}
</style>
