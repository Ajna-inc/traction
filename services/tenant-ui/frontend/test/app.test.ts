import { expect, test, vi } from 'vitest';

import App from '../src/App.vue';
import { mount } from '@vue/test-utils';

vi.mock('pinia', () => ({
  defineStore: vi.fn(),
  storeToRefs: vi.fn(() => ({
    config: {
      value: {
        frontend: {
          ux: {
            appTitle: 'Tenant UI',
          },
        },
      },
    },
  })),
}));
vi.mock('@/store/configStore', () => ({
  useConfigStore: vi.fn(),
}));

test('mount App', () => {
  const wrapper = mount(App);

  expect(App).toBeTruthy();
  expect(wrapper.html()).toMatchSnapshot();
});
