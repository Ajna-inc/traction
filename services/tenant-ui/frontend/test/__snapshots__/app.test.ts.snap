// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount App 1`] = `
"<router-view data-v-7a7a37b1=\\"\\"></router-view>
<!-- Shared confirm popup  -->
<!---->"
`;
